src/unix/thread.c

```diff
@@ -72,10 +75,59 @@ int uv_thread_create(uv_thread_t *tid, void (*entry)(void *arg), void *arg) {
 }
 
 
+int uv_thread_create_on(int proc_num,
+                        uv_thread_t *tid,
+                        void (*entry)(void *arg),
+                        void *arg) {
```

Comment:
A versioned API is no problem--I understand the need better now.

But, on further thought, this is beginning to look a bit like a rabbit hole. Considering that a number of the OSes you listed above only have post-creation affinity setting capability, what do you think of dropping ``uv_thread_create_with_attr()`` and only adding the following:

```
int uv_thread_setaffinity(uv_thread_t *tid, char *cpumask, size_t cpumask_size);
int uv_thread_getaffinity(uv_thread_t *tid, char cpumask[UV_MAX_CPUMASK_SIZE], size_t *cpumask_size);
```